pkg/conduit/config.go

```diff
@@ -39,47 +39,46 @@ const (
 	SchemaRegistryTypeBuiltin   = "builtin"
 )
 
-type ConfigDB struct {
-	// When Driver is specified it takes precedence over other DB related
-	// fields.
-	Driver database.DB
-
-	Type   string `long:"db.type" usage:"database type; accepts badger,postgres,inmemory,sqlite"`
-	Badger struct {
-		Path string `long:"db.badger.path" usage:"path to badger DB"`
-	}
-	Postgres struct {
-		ConnectionString string `long:"db.postgres.connection-string" usage:"postgres connection string, may be a database URL or in PostgreSQL keyword/value format"`
-		Table            string `long:"db.postgres.table" usage:"postgres table in which to store data (will be created if it does not exist)"`
-	}
-	SQLite struct {
-		Path  string `long:"db.sqlite.path" usage:"path to sqlite3 DB"`
-		Table string `long:"db.sqlite.table" usage:"sqlite3 table in which to store data (will be created if it does not exist)"`
-	}
-}
+// Config holds all configurable values for Conduit.
+type Config struct {
+	ConduitCfgPath string `long:"config.path" usage:"global conduit configuration file" default:"./conduit.yaml"`
 
-type ConfigAPI struct {
-	Enabled bool `long:"api.enabled" usage:"enable HTTP and gRPC API"`
-	HTTP    struct {
-		Address string `long:"http.address" usage:"address for serving the HTTP API"`
-	}
-	GRPC struct {
-		Address string `long:"grpc.address" usage:"address for serving the gRPC API"`
+	DB struct {
+		// When Driver is specified it takes precedence over other DB related
+		// fields.
+		Driver database.DB
+
+		Type   string `long:"db.type" usage:"database type; accepts badger,postgres,inmemory,sqlite"`
+		Badger struct {
+			Path string `long:"db.badger.path" usage:"path to badger DB"`
+		}
+		Postgres struct {
+			ConnectionString string `long:"db.postgres.connection-string" usage:"postgres connection string, may be a database URL or in PostgreSQL keyword/value format"`
```

Comment:
about `connection-string`, does camel case not work for a command? or was this format agreed on by the team?

Reply:
@maha-hajja `connection-string` was already a conduit flag and part of our configuration files https://conduit.io/docs/configuration#cli-flags. It's how all our flags with two words are normally named.

Reply:
never noticed it for some reason! thanks for the reference